crates/router/src/core/payments/transformers.rs

```diff
@@ -136,6 +137,9 @@ where
         integrity_check: Ok(()),
         additional_merchant_data: None,
         header_payload: None,
+        connector_mandate_request_reference_id: Some(common_utils::generate_id_with_len(
+            consts::CONNECTOR_MANDATE_REQUEST_REFERENCE_ID_LENGTH.to_owned(),
+        )),
```

Comment:
Same here, generating a random ID and sharing it with the connector, but not storing it on our database is useless. We won't have a track of the value we'd sent to the connector, and we can't take any further actions where we need the value.

We should ideally avoid such usages in the first place.